{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of PEAKS de novo results of Trocas 8 (low water, November 2019) non-incubation samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    PEAKS de novo results (.csv) of PTM-optimized sequencing >50% ALC\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the TROCAS # (e.g., 675) for another\n",
    "#### 2. Change output file name in *IN 6*, *IN 7*, *IN 8*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/T8-June2021-UWPR-DN50_10lgP15/682_Trocas8_June2021_DENOVO_109\n"
     ]
    }
   ],
   "source": [
    "    cd /home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/T8-June2021-UWPR-DN50_10lgP15/682_Trocas8_June2021_DENOVO_109/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Peaks peptides >50% ALC in combined dataframe: 46\n",
      "Index(['Fraction', 'Scan', 'Source File', 'Peptide', 'Tag Length', 'ALC (%)',\n",
      "       'length', 'm/z', 'z', 'RT', 'Area', 'Mass', 'ppm', 'PTM',\n",
      "       'local confidence (%)', 'tag (>=0%)', 'mode'],\n",
      "      dtype='object')\n",
      "8.782608695652174\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>ALC</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>98</td>\n",
       "      <td>10</td>\n",
       "      <td>523.2850</td>\n",
       "      <td>2</td>\n",
       "      <td>44.70</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>1044.5564</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>98</td>\n",
       "      <td>10</td>\n",
       "      <td>523.2850</td>\n",
       "      <td>2</td>\n",
       "      <td>45.21</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>1044.5564</td>\n",
       "      <td>-0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>98</td>\n",
       "      <td>10</td>\n",
       "      <td>523.2850</td>\n",
       "      <td>2</td>\n",
       "      <td>44.20</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>1044.5564</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YLYELAR</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>464.2501</td>\n",
       "      <td>2</td>\n",
       "      <td>69.21</td>\n",
       "      <td>17700.0</td>\n",
       "      <td>926.4861</td>\n",
       "      <td>-0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VATVSPLR</td>\n",
       "      <td>95</td>\n",
       "      <td>8</td>\n",
       "      <td>421.7586</td>\n",
       "      <td>2</td>\n",
       "      <td>57.07</td>\n",
       "      <td>12700000.0</td>\n",
       "      <td>841.5021</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Peptide  ALC  length       m/z  z     RT        Area       Mass  ppm\n",
       "0  LSSPATLNSR   98      10  523.2850  2  44.70   5110000.0  1044.5564 -1.0\n",
       "1  LSSPATLNSR   98      10  523.2850  2  45.21   5110000.0  1044.5564 -0.8\n",
       "2  LSSPATLNSR   98      10  523.2850  2  44.20   5110000.0  1044.5564 -1.0\n",
       "3     YLYELAR   98       7  464.2501  2  69.21     17700.0   926.4861 -0.6\n",
       "4    VATVSPLR   95       8  421.7586  2  57.07  12700000.0   841.5021  0.6"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSVs into a dataframe using the read_csv function and call 'peaks'\n",
    "\n",
    "peaks682 = pd.read_csv(\"/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/T8-June2021-UWPR-DN50_10lgP15/682_Trocas8_June2021_DENOVO_109/682_Trocas8_June2021_DENOVO_109_DN50.csv\")\n",
    "\n",
    "print(\"# redundant Peaks peptides >50% ALC in combined dataframe:\", len(peaks682))\n",
    "\n",
    "print(peaks682.columns)\n",
    "\n",
    "# These columns mess things up- get rid of them\n",
    "\n",
    "del peaks682['Fraction']\n",
    "del peaks682['Scan']\n",
    "del peaks682['Source File']\n",
    "del peaks682['Tag Length']\n",
    "del peaks682['PTM']\n",
    "del peaks682['tag (>=0%)']\n",
    "del peaks682['mode']\n",
    "del peaks682['local confidence (%)']\n",
    "\n",
    "\n",
    "columns = ['Peptide', 'ALC', 'length', 'm/z', 'z', 'RT', 'Area',\n",
    "       'Mass', 'ppm']\n",
    "\n",
    "peaks682.columns = columns\n",
    "\n",
    "mean_len = peaks682['length'].mean()\n",
    "print(mean_len)\n",
    "\n",
    "# look at the dataframe\n",
    "peaks682.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The peptide column has the masses of modifications (e.g., 57.02 Da for carbamidomethylation of cysteine). We want to make new columns with all that information and make a column with only the 'stripped' peptide sequence that's just amino acids - this we can then align against other sequences, for example.\n",
    "\n",
    "Modified residues were allowed for:\n",
    "\n",
    "    fixed carbamodimethylation of cysteine 57.021464 C\n",
    "    varialbe oxidation of methionine: 15.9949 M\n",
    "    variable deamidation of asparagine, glumatine: 0.984016 NQ\n",
    "\n",
    "We'll then write this manipulated dataframe to a new file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>ALC</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "      <th>A</th>\n",
       "      <th>...</th>\n",
       "      <th>W</th>\n",
       "      <th>Y</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>stripped_peptide</th>\n",
       "      <th>stripped_length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF_num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>98</td>\n",
       "      <td>10</td>\n",
       "      <td>523.2850</td>\n",
       "      <td>2</td>\n",
       "      <td>44.70</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>1044.5564</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>5.110000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>98</td>\n",
       "      <td>10</td>\n",
       "      <td>523.2850</td>\n",
       "      <td>2</td>\n",
       "      <td>45.21</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>1044.5564</td>\n",
       "      <td>-0.8</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>5.110000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>98</td>\n",
       "      <td>10</td>\n",
       "      <td>523.2850</td>\n",
       "      <td>2</td>\n",
       "      <td>44.20</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>1044.5564</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>5.110000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YLYELAR</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>464.2501</td>\n",
       "      <td>2</td>\n",
       "      <td>69.21</td>\n",
       "      <td>17700.0</td>\n",
       "      <td>926.4861</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>YLYELAR</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2.528571e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VATVSPLR</td>\n",
       "      <td>95</td>\n",
       "      <td>8</td>\n",
       "      <td>421.7586</td>\n",
       "      <td>2</td>\n",
       "      <td>57.07</td>\n",
       "      <td>12700000.0</td>\n",
       "      <td>841.5021</td>\n",
       "      <td>0.6</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VATVSPLR</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1.587500e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Peptide  ALC  length       m/z  z     RT        Area       Mass  ppm  A  \\\n",
       "0  LSSPATLNSR   98      10  523.2850  2  44.70   5110000.0  1044.5564 -1.0  1   \n",
       "1  LSSPATLNSR   98      10  523.2850  2  45.21   5110000.0  1044.5564 -0.8  1   \n",
       "2  LSSPATLNSR   98      10  523.2850  2  44.20   5110000.0  1044.5564 -1.0  1   \n",
       "3     YLYELAR   98       7  464.2501  2  69.21     17700.0   926.4861 -0.6  1   \n",
       "4    VATVSPLR   95       8  421.7586  2  57.07  12700000.0   841.5021  0.6  1   \n",
       "\n",
       "   ...  W  Y  c-carb  m-oxid  n-deam  q-deam  stripped_peptide  \\\n",
       "0  ...  0  0       0       0       0       0        LSSPATLNSR   \n",
       "1  ...  0  0       0       0       0       0        LSSPATLNSR   \n",
       "2  ...  0  0       0       0       0       0        LSSPATLNSR   \n",
       "3  ...  0  2       0       0       0       0           YLYELAR   \n",
       "4  ...  0  0       0       0       0       0          VATVSPLR   \n",
       "\n",
       "   stripped_length  ptm-total     NAAF_num.  \n",
       "0               10          0  5.110000e+05  \n",
       "1               10          0  5.110000e+05  \n",
       "2               10          0  5.110000e+05  \n",
       "3                7          0  2.528571e+03  \n",
       "4                8          0  1.587500e+06  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks682['A'] = peaks682['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks682['C'] = peaks682['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks682['D'] = peaks682['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks682['E'] = peaks682['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks682['F'] = peaks682['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks682['G'] = peaks682['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks682['H'] = peaks682['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks682 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks682['I'] = peaks682['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks682['K'] = peaks682['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks682['L'] = peaks682['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks682['M'] = peaks682['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks682['N'] = peaks682['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks682['P'] = peaks682['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks682['Q'] = peaks682['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks682['R'] = peaks682['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks682['S'] = peaks682['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks682['T'] = peaks682['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks682['V'] = peaks682['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks682['W'] = peaks682['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks682['Y'] = peaks682['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks682['c-carb'] = peaks682['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks682['m-oxid'] = peaks682['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks682['n-deam'] = peaks682['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks682['q-deam'] = peaks682['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks682['stripped_peptide'] = peaks682['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks682['stripped_length'] = peaks682['stripped_peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks682['ptm-total'] = peaks682['c-carb'] + peaks682['m-oxid'] + peaks682['n-deam'] + peaks682['q-deam']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks682['NAAF_num.'] = peaks682['Area'] / peaks682['stripped_length']\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "peaks682.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/PeaksDN/682_NMCP_surf_GF75_DN/682_NMCP_surf_GF75_DN50.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaks682.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped_peptide</th>\n",
       "      <th>Area</th>\n",
       "      <th>NAAF_num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>5.110000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>5.110000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>5110000.0</td>\n",
       "      <td>5.110000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YLYELAR</td>\n",
       "      <td>17700.0</td>\n",
       "      <td>2.528571e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VATVSPLR</td>\n",
       "      <td>12700000.0</td>\n",
       "      <td>1.587500e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped_peptide        Area     NAAF_num.\n",
       "0       LSSPATLNSR   5110000.0  5.110000e+05\n",
       "1       LSSPATLNSR   5110000.0  5.110000e+05\n",
       "2       LSSPATLNSR   5110000.0  5.110000e+05\n",
       "3          YLYELAR     17700.0  2.528571e+03\n",
       "4         VATVSPLR  12700000.0  1.587500e+06"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only stripped peptide I/L and NAAF\n",
    "dn_682 = peaks682[['stripped_peptide', 'Area', 'NAAF_num.']]\n",
    "\n",
    "dn_682.set_index('stripped_peptide')\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "dn_682.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/I-L_NAAFs/682_NMCP_surf_GF75_DN50_ILnaafs.csv\")\n",
    "\n",
    "dn_682.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>55</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.073589e+08</td>\n",
       "      <td>402</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.066667</td>\n",
       "      <td>0.0</td>\n",
       "      <td>267062.034776</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               A  C  D   E  F   G   H  I   K   L  ...  m-oxid  n-deam  q-deam  \\\n",
       "sample total  23  1  2  14  3  11  21  0  13  55  ...       7       1       0   \n",
       "\n",
       "                Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  1.073589e+08           402          1.0         0.35   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam    NAAF denom.  \n",
       "sample total     0.066667          0.0  267062.034776  \n",
       "\n",
       "[1 rows x 31 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns \n",
    "# in the stripped peptide dataframe above (for >50% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks682['A'].sum(),\n",
    "        'C': peaks682['C'].sum(),\n",
    "        'D': peaks682['D'].sum(),\n",
    "        'E': peaks682['E'].sum(),\n",
    "        'F': peaks682['F'].sum(),\n",
    "        'G': peaks682['G'].sum(),\n",
    "        'H': peaks682['H'].sum(),\n",
    "        'I': peaks682['I'].sum(),\n",
    "        'K': peaks682['K'].sum(),\n",
    "        'L': peaks682['L'].sum(),\n",
    "        'M': peaks682['M'].sum(),\n",
    "        'N': peaks682['N'].sum(),\n",
    "        'P': peaks682['P'].sum(),\n",
    "        'Q': peaks682['Q'].sum(),\n",
    "        'R': peaks682['R'].sum(),\n",
    "        'S': peaks682['S'].sum(),\n",
    "        'T': peaks682['T'].sum(),\n",
    "        'V': peaks682['V'].sum(),\n",
    "        'W': peaks682['W'].sum(),\n",
    "        'Y': peaks682['Y'].sum(),\n",
    "        'c-carb': peaks682['c-carb'].sum(),\n",
    "        'm-oxid': peaks682['m-oxid'].sum(),\n",
    "        'n-deam': peaks682['n-deam'].sum(),\n",
    "        'q-deam': peaks682['q-deam'].sum(),\n",
    "        'Total area': peaks682['Area'].sum(),\n",
    "        'Total length': peaks682['stripped_length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks682 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', \\\n",
    "                                            'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', \\\n",
    "                                            'c-carb', 'm-oxid', 'n-deam', 'q-deam', \\\n",
    "                                            'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks682['% C w/ carb'] = totalpeaks682['c-carb'] / totalpeaks682['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks682['% M w/ oxid'] = totalpeaks682['m-oxid'] / totalpeaks682['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks682['% N w/ deam'] = totalpeaks682['n-deam'] / totalpeaks682['N'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks682['% Q w/ deam'] = totalpeaks682['q-deam'] / totalpeaks682['Q'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks682['NAAF denom.'] = totalpeaks682['Total area'] / totalpeaks682['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks682.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/PeaksDN/682_NMCP_surf_GF75_DN/682_NMCP_surf_GF75_DN50_totals.csv\")\n",
    "\n",
    "totalpeaks682.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped_peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>K</th>\n",
       "      <th>...</th>\n",
       "      <th>R-NAAF50</th>\n",
       "      <th>S-NAAF50</th>\n",
       "      <th>T-NAAF50</th>\n",
       "      <th>V-NAAF50</th>\n",
       "      <th>W-NAAF50</th>\n",
       "      <th>Y-NAAF50</th>\n",
       "      <th>ccarb-NAAF50</th>\n",
       "      <th>moxid-NAAF50</th>\n",
       "      <th>ndeam-NAAF50</th>\n",
       "      <th>qdeam-NAAF50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>5.740239</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>5.740239</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>5.740239</td>\n",
       "      <td>1.913413</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YLYELAR</td>\n",
       "      <td>0.009468</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.009468</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.018936</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VATVSPLR</td>\n",
       "      <td>5.944312</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.944312</td>\n",
       "      <td>5.944312</td>\n",
       "      <td>5.944312</td>\n",
       "      <td>11.888624</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped_peptide  NAAF factor  A  C  D  E  F  G  H  K  ...  R-NAAF50  \\\n",
       "0       LSSPATLNSR     1.913413  1  0  0  0  0  0  0  0  ...  1.913413   \n",
       "1       LSSPATLNSR     1.913413  1  0  0  0  0  0  0  0  ...  1.913413   \n",
       "2       LSSPATLNSR     1.913413  1  0  0  0  0  0  0  0  ...  1.913413   \n",
       "3          YLYELAR     0.009468  1  0  0  1  0  0  0  0  ...  0.009468   \n",
       "4         VATVSPLR     5.944312  1  0  0  0  0  0  0  0  ...  5.944312   \n",
       "\n",
       "   S-NAAF50  T-NAAF50   V-NAAF50  W-NAAF50  Y-NAAF50  ccarb-NAAF50  \\\n",
       "0  5.740239  1.913413   0.000000       0.0  0.000000           0.0   \n",
       "1  5.740239  1.913413   0.000000       0.0  0.000000           0.0   \n",
       "2  5.740239  1.913413   0.000000       0.0  0.000000           0.0   \n",
       "3  0.000000  0.000000   0.000000       0.0  0.018936           0.0   \n",
       "4  5.944312  5.944312  11.888624       0.0  0.000000           0.0   \n",
       "\n",
       "   moxid-NAAF50  ndeam-NAAF50  qdeam-NAAF50  \n",
       "0           0.0           0.0           0.0  \n",
       "1           0.0           0.0           0.0  \n",
       "2           0.0           0.0           0.0  \n",
       "3           0.0           0.0           0.0  \n",
       "4           0.0           0.0           0.0  \n",
       "\n",
       "[5 rows x 50 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we would with Comet results\n",
    "\n",
    "NAAF50 = 267062.034776\n",
    "\n",
    "# use NAAF >50% ALC to get NAAF factor\n",
    "peaks682['NAAF factor'] = (peaks682['NAAF_num.'])/NAAF50\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaksAAPTM_682 = peaks682[['stripped_peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'K', 'I', 'L', \\\n",
    "                                'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', \\\n",
    "                                'n-deam', 'q-deam']].copy()\n",
    "\n",
    "# multiply the NAAF50 factor by the AA total to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksAAPTM_682['A-NAAF50'] = peaksAAPTM_682['A'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['C-NAAF50'] = peaksAAPTM_682['C'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['D-NAAF50'] = peaksAAPTM_682['D'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['E-NAAF50'] = peaksAAPTM_682['E'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['F-NAAF50'] = peaksAAPTM_682['F'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['G-NAAF50'] = peaksAAPTM_682['G'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['H-NAAF50'] = peaksAAPTM_682['H'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['I-NAAF50'] = peaksAAPTM_682['I'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['K-NAAF50'] = peaksAAPTM_682['K'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['L-NAAF50'] = peaksAAPTM_682['L'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['M-NAAF50'] = peaksAAPTM_682['M'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['N-NAAF50'] = peaksAAPTM_682['N'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['P-NAAF50'] = peaksAAPTM_682['P'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['Q-NAAF50'] = peaksAAPTM_682['Q'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['R-NAAF50'] = peaksAAPTM_682['R'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['S-NAAF50'] = peaksAAPTM_682['S'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['T-NAAF50'] = peaksAAPTM_682['T'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['V-NAAF50'] = peaksAAPTM_682['V'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['W-NAAF50'] = peaksAAPTM_682['W'] * peaks682['NAAF factor']\n",
    "peaksAAPTM_682['Y-NAAF50'] = peaksAAPTM_682['Y'] * peaks682['NAAF factor']\n",
    "\n",
    "# multiply the NAAF50 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksAAPTM_682['ccarb-NAAF50'] = peaksAAPTM_682['c-carb'] * peaksAAPTM_682['NAAF factor']\n",
    "peaksAAPTM_682['moxid-NAAF50'] = peaksAAPTM_682['m-oxid'] * peaksAAPTM_682['NAAF factor']\n",
    "peaksAAPTM_682['ndeam-NAAF50'] = peaksAAPTM_682['n-deam'] * peaksAAPTM_682['NAAF factor']\n",
    "peaksAAPTM_682['qdeam-NAAF50'] = peaksAAPTM_682['q-deam'] * peaksAAPTM_682['NAAF factor']\n",
    "\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaksAAPTM_682.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/PeaksDN/682_NMCP_surf_GF75_DN/682_NMCP_surf_GF75_DN50_naaf.csv\")\n",
    "\n",
    "peaksAAPTM_682.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>...</th>\n",
       "      <th>Y</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>% C w/ carb.</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>48.707006</td>\n",
       "      <td>33.122496</td>\n",
       "      <td>0.014678</td>\n",
       "      <td>0.167252</td>\n",
       "      <td>0.317797</td>\n",
       "      <td>0.108637</td>\n",
       "      <td>2.840398</td>\n",
       "      <td>0.474454</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.334068</td>\n",
       "      <td>...</td>\n",
       "      <td>0.257652</td>\n",
       "      <td>0.014678</td>\n",
       "      <td>0.121538</td>\n",
       "      <td>0.151026</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.224636</td>\n",
       "      <td>0.02493</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000182</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   NAAF          A         C         D         E         F  \\\n",
       "sample total  48.707006  33.122496  0.014678  0.167252  0.317797  0.108637   \n",
       "\n",
       "                     G         H    I         K  ...         Y    c-carb  \\\n",
       "sample total  2.840398  0.474454  0.0  0.334068  ...  0.257652  0.014678   \n",
       "\n",
       "                m-oxid    n-deam  q-deam  % C w/ carb.  % M w/ oxid  \\\n",
       "sample total  0.121538  0.151026     0.0           1.0     0.224636   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  NAAF check  \n",
       "sample total      0.02493          0.0    0.000182  \n",
       "\n",
       "[1 rows x 30 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a dataframe that's the sum of NAAF corrected AAs and PTMs\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaksAAPTM_682['NAAF factor'].sum(),\n",
    "        'A': peaksAAPTM_682['A-NAAF50'].sum(),\n",
    "        'C': peaksAAPTM_682['C-NAAF50'].sum(),\n",
    "        'D': peaksAAPTM_682['D-NAAF50'].sum(),\n",
    "        'E': peaksAAPTM_682['E-NAAF50'].sum(),\n",
    "        'F': peaksAAPTM_682['F-NAAF50'].sum(),\n",
    "        'G': peaksAAPTM_682['G-NAAF50'].sum(),\n",
    "        'H': peaksAAPTM_682['H-NAAF50'].sum(),\n",
    "        'I': peaksAAPTM_682['I-NAAF50'].sum(),\n",
    "        'K': peaksAAPTM_682['K-NAAF50'].sum(),\n",
    "        'L': peaksAAPTM_682['L-NAAF50'].sum(),\n",
    "        'M': peaksAAPTM_682['M-NAAF50'].sum(),\n",
    "        'N': peaksAAPTM_682['N-NAAF50'].sum(),\n",
    "        'P': peaksAAPTM_682['P-NAAF50'].sum(),\n",
    "        'Q': peaksAAPTM_682['Q-NAAF50'].sum(),\n",
    "        'R': peaksAAPTM_682['R-NAAF50'].sum(),\n",
    "        'S': peaksAAPTM_682['S-NAAF50'].sum(),\n",
    "        'T': peaksAAPTM_682['T-NAAF50'].sum(),\n",
    "        'V': peaksAAPTM_682['V-NAAF50'].sum(),\n",
    "        'W': peaksAAPTM_682['W-NAAF50'].sum(),\n",
    "        'Y': peaksAAPTM_682['Y-NAAF50'].sum(),\n",
    "        'c-carb': peaksAAPTM_682['ccarb-NAAF50'].sum(),\n",
    "        'm-oxid': peaksAAPTM_682['moxid-NAAF50'].sum(),\n",
    "        'n-deam': peaksAAPTM_682['ndeam-NAAF50'].sum(),\n",
    "        'q-deam': peaksAAPTM_682['qdeam-NAAF50'].sum(),\n",
    "       }\n",
    "\n",
    "totalpeaks50_NAAF = pd.DataFrame(data, columns=['NAAF', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', \\\n",
    "                                           'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', \\\n",
    "                                           'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam'], index=index)\n",
    "\n",
    "# calculate NAAF-corrected percentage of C's with carb (should be 1.0)\n",
    "totalpeaks50_NAAF['% C w/ carb.'] = totalpeaks50_NAAF['c-carb'] / totalpeaks50_NAAF['C'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of M's that are oxidized\n",
    "totalpeaks50_NAAF['% M w/ oxid'] = totalpeaks50_NAAF['m-oxid'] / totalpeaks50_NAAF['M'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of N's that are deamidated\n",
    "totalpeaks50_NAAF['% N w/ deam'] = totalpeaks50_NAAF['n-deam'] / totalpeaks50_NAAF['N'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of N's that are deamidated\n",
    "totalpeaks50_NAAF['% Q w/ deam'] = totalpeaks50_NAAF['q-deam'] / totalpeaks50_NAAF['Q'] \n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks50_NAAF['NAAF check'] = totalpeaks50_NAAF['NAAF'] / 267062.034776\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks50_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/PeaksDN/682_NMCP_surf_GF75_DN/682_NMCP_surf_GF75_DN50_naaf_totals.csv\")\n",
    "\n",
    "totalpeaks50_NAAF.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export stripped peptides >50% ALC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant stripped Peaks peptides >50% ALC 46\n",
      "# nonredundant stripped Peaks peptides >50% ALC 34\n",
      "average peptide length Peaks peptides >50% ALC 8.73913043478261\n",
      "# redundant Peaks peptides >50% ALC 46\n",
      "# nonredundant Peaks peptides 34\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped_peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LSSPATLNSR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YLYELAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VATVSPLR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped_peptide\n",
       "0       LSSPATLNSR\n",
       "1       LSSPATLNSR\n",
       "2       LSSPATLNSR\n",
       "3          YLYELAR\n",
       "4         VATVSPLR"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###### keep only stripped peptide column \n",
    "pep50 = peaks682[[\"stripped_peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep50.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/PeaksDN/682_NMCP_surf_GF75_DN/682_NMCP_surf_GF75_DN50_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/PeaksDN/682_NMCP_surf_GF75_DN/682_NMCP_surf_GF75_DN50_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/amazon/data/Trocas8-notincs/processed/PeaksDN/682_NMCP_surf_GF75_DN/682_NMCP_surf_GF75_DN50_stripped_peptides.fas\n",
    "\n",
    "# removing redundancy\n",
    "peaks50dedup = pd.DataFrame.drop_duplicates(pep50)\n",
    "\n",
    "print(\"# redundant stripped Peaks peptides >50% ALC\", len(pep50))\n",
    "print(\"# nonredundant stripped Peaks peptides >50% ALC\", len(peaks50dedup))\n",
    "print(\"average peptide length Peaks peptides >50% ALC\", peaks682['stripped_length'].mean())\n",
    "\n",
    "# count all unique peptide (modified peptides included)\n",
    "# keep only peptide column >50% ALC\n",
    "pep50m = peaks682[[\"Peptide\"]]\n",
    "\n",
    "# deduplicate\n",
    "pep50mdedup = pd.DataFrame.drop_duplicates(pep50m)\n",
    "\n",
    "print(\"# redundant Peaks peptides >50% ALC\", len(pep50m))\n",
    "print(\"# nonredundant Peaks peptides\", len(pep50mdedup))\n",
    "\n",
    "# check\n",
    "pep50.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
