{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of PEAKS de novo results of Trocas7 (April, 2019 high water) lower Amazon River proteomics LC-MS/MS data using Python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "PEAKS 8.5 de novo results (.csv) of all combined samples sequencing >580% ALC\n",
    "from Thermo Fusion tribrid runs at the UW Proteomics Resource center, April 2021\n",
    "combined from multiple injections\n",
    "\n",
    "Goal:\n",
    "\n",
    "Files with stripped (no PTMs) peptide lists and\n",
    "Columns with #'s of each modification in every sequence\n",
    "Column with stripped peptide lengths (# amino acids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/amazon/data/TROCAS7_Fusion_Apr2021_DENOVO-75-all-samples\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/amazon/data/TROCAS7_Fusion_Apr2021_DENOVO-75-all-samples/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Apr21-all-DN50.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Peaks peptides >50% ALC in combined dataframe: 7201\n",
      "Index(['Fraction', 'Scan', 'Source File', 'Peptide', 'Tag Length', 'ALC (%)',\n",
      "       'length', 'm/z', 'z', 'RT', 'Area', 'Mass', 'ppm', 'PTM',\n",
      "       'local confidence (%)', 'tag (>=0%)', 'mode'],\n",
      "      dtype='object')\n",
      "10.145535342313568\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>ALC</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EC(+57.02)ELSC(+57.02)K</td>\n",
       "      <td>99</td>\n",
       "      <td>7</td>\n",
       "      <td>463.1921</td>\n",
       "      <td>2</td>\n",
       "      <td>37.66</td>\n",
       "      <td>70400.0</td>\n",
       "      <td>924.3681</td>\n",
       "      <td>1.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VLEGNEQFLNAAK</td>\n",
       "      <td>99</td>\n",
       "      <td>13</td>\n",
       "      <td>716.8768</td>\n",
       "      <td>2</td>\n",
       "      <td>75.43</td>\n",
       "      <td>3970000.0</td>\n",
       "      <td>1431.7358</td>\n",
       "      <td>2.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VLEGNEQFLNAAK</td>\n",
       "      <td>99</td>\n",
       "      <td>13</td>\n",
       "      <td>716.8758</td>\n",
       "      <td>2</td>\n",
       "      <td>71.30</td>\n",
       "      <td>618000.0</td>\n",
       "      <td>1431.7358</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SC(+57.02)ELSC(+57.02)K</td>\n",
       "      <td>99</td>\n",
       "      <td>7</td>\n",
       "      <td>442.1857</td>\n",
       "      <td>2</td>\n",
       "      <td>22.79</td>\n",
       "      <td>242000.0</td>\n",
       "      <td>882.3575</td>\n",
       "      <td>-0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EC(+57.02)ELSC(+57.02)K</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>463.1911</td>\n",
       "      <td>2</td>\n",
       "      <td>37.47</td>\n",
       "      <td>48500.0</td>\n",
       "      <td>924.3681</td>\n",
       "      <td>-0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Peptide  ALC  length       m/z  z     RT       Area  \\\n",
       "0  EC(+57.02)ELSC(+57.02)K   99       7  463.1921  2  37.66    70400.0   \n",
       "1            VLEGNEQFLNAAK   99      13  716.8768  2  75.43  3970000.0   \n",
       "2            VLEGNEQFLNAAK   99      13  716.8758  2  71.30   618000.0   \n",
       "3  SC(+57.02)ELSC(+57.02)K   99       7  442.1857  2  22.79   242000.0   \n",
       "4  EC(+57.02)ELSC(+57.02)K   98       7  463.1911  2  37.47    48500.0   \n",
       "\n",
       "        Mass  ppm  \n",
       "0   924.3681  1.6  \n",
       "1  1431.7358  2.3  \n",
       "2  1431.7358  0.8  \n",
       "3   882.3575 -0.7  \n",
       "4   924.3681 -0.4  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSVs into a dataframe using the read_csv function and call 'peaks'\n",
    "\n",
    "peaks = pd.read_csv(\"/home/millieginty/Documents/git-repos/amazon/data/TROCAS7_Fusion_Apr2021_DENOVO-75-all-samples/Apr21-all-DN50.csv\")\n",
    "\n",
    "print(\"# redundant Peaks peptides >50% ALC in combined dataframe:\", len(peaks))\n",
    "\n",
    "print(peaks.columns)\n",
    "\n",
    "# These columns mess things up- get rid of them\n",
    "\n",
    "del peaks['Fraction']\n",
    "del peaks['Scan']\n",
    "del peaks['Source File']\n",
    "del peaks['Tag Length']\n",
    "del peaks['PTM']\n",
    "del peaks['tag (>=0%)']\n",
    "del peaks['mode']\n",
    "del peaks['local confidence (%)']\n",
    "\n",
    "\n",
    "columns = ['Peptide', 'ALC', 'length', 'm/z', 'z', 'RT', 'Area',\n",
    "       'Mass', 'ppm']\n",
    "\n",
    "peaks.columns = columns\n",
    "\n",
    "mean_len = peaks['length'].mean()\n",
    "print(mean_len)\n",
    "\n",
    "# look at the dataframe\n",
    "peaks.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>ALC</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "      <th>A</th>\n",
       "      <th>...</th>\n",
       "      <th>W</th>\n",
       "      <th>Y</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>stripped_peptide</th>\n",
       "      <th>stripped_length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF_num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EC(+57.02)ELSC(+57.02)K</td>\n",
       "      <td>99</td>\n",
       "      <td>7</td>\n",
       "      <td>463.1921</td>\n",
       "      <td>2</td>\n",
       "      <td>37.66</td>\n",
       "      <td>70400.0</td>\n",
       "      <td>924.3681</td>\n",
       "      <td>1.6</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ECK</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>23466.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VLEGNEQFLNAAK</td>\n",
       "      <td>99</td>\n",
       "      <td>13</td>\n",
       "      <td>716.8768</td>\n",
       "      <td>2</td>\n",
       "      <td>75.43</td>\n",
       "      <td>3970000.0</td>\n",
       "      <td>1431.7358</td>\n",
       "      <td>2.3</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VLEGNEQFLNAAK</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>305384.615385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VLEGNEQFLNAAK</td>\n",
       "      <td>99</td>\n",
       "      <td>13</td>\n",
       "      <td>716.8758</td>\n",
       "      <td>2</td>\n",
       "      <td>71.30</td>\n",
       "      <td>618000.0</td>\n",
       "      <td>1431.7358</td>\n",
       "      <td>0.8</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VLEGNEQFLNAAK</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>47538.461538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SC(+57.02)ELSC(+57.02)K</td>\n",
       "      <td>99</td>\n",
       "      <td>7</td>\n",
       "      <td>442.1857</td>\n",
       "      <td>2</td>\n",
       "      <td>22.79</td>\n",
       "      <td>242000.0</td>\n",
       "      <td>882.3575</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SCK</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>80666.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EC(+57.02)ELSC(+57.02)K</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>463.1911</td>\n",
       "      <td>2</td>\n",
       "      <td>37.47</td>\n",
       "      <td>48500.0</td>\n",
       "      <td>924.3681</td>\n",
       "      <td>-0.4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ECK</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>16166.666667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Peptide  ALC  length       m/z  z     RT       Area  \\\n",
       "0  EC(+57.02)ELSC(+57.02)K   99       7  463.1921  2  37.66    70400.0   \n",
       "1            VLEGNEQFLNAAK   99      13  716.8768  2  75.43  3970000.0   \n",
       "2            VLEGNEQFLNAAK   99      13  716.8758  2  71.30   618000.0   \n",
       "3  SC(+57.02)ELSC(+57.02)K   99       7  442.1857  2  22.79   242000.0   \n",
       "4  EC(+57.02)ELSC(+57.02)K   98       7  463.1911  2  37.47    48500.0   \n",
       "\n",
       "        Mass  ppm  A  ...  W  Y  c-carb  m-oxid  n-deam  q-deam  \\\n",
       "0   924.3681  1.6  0  ...  0  0       2       0       0       0   \n",
       "1  1431.7358  2.3  2  ...  0  0       0       0       0       0   \n",
       "2  1431.7358  0.8  2  ...  0  0       0       0       0       0   \n",
       "3   882.3575 -0.7  0  ...  0  0       2       0       0       0   \n",
       "4   924.3681 -0.4  0  ...  0  0       2       0       0       0   \n",
       "\n",
       "   stripped_peptide  stripped_length  ptm-total      NAAF_num.  \n",
       "0               ECK                3          2   23466.666667  \n",
       "1     VLEGNEQFLNAAK               13          0  305384.615385  \n",
       "2     VLEGNEQFLNAAK               13          0   47538.461538  \n",
       "3               SCK                3          2   80666.666667  \n",
       "4               ECK                3          2   16166.666667  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks['A'] = peaks['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks['C'] = peaks['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks['D'] = peaks['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks['E'] = peaks['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks['F'] = peaks['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks['G'] = peaks['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks['H'] = peaks['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks['I'] = peaks['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks['K'] = peaks['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks['L'] = peaks['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks['M'] = peaks['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks['N'] = peaks['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks['P'] = peaks['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks['Q'] = peaks['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks['R'] = peaks['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks['S'] = peaks['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks['T'] = peaks['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks['V'] = peaks['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks['W'] = peaks['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks['Y'] = peaks['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks['c-carb'] = peaks['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks['m-oxid'] = peaks['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized K's in each peptide\n",
    "#peaks['k-oxid'] = peaks['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized P's in each peptide\n",
    "#peaks['p-oxid'] = peaks['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized R's in each peptide\n",
    "#peaks['r-oxid'] = peaks['Peptide'].apply(lambda x: x.count('R(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized Y's in each peptide\n",
    "#peaks['y-oxid'] = peaks['Peptide'].apply(lambda x: x.count('Y(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks['n-deam'] = peaks['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks['q-deam'] = peaks['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "#peaks['k-meth'] = peaks['Peptide'].apply(lambda x: x.count('K(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "#peaks['r-meth'] = peaks['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of pyro glu Q's in each peptide\n",
    "#peaks['q-pyro'] = peaks['Peptide'].apply(lambda x: x.count('Q(-17.03)'))\n",
    "\n",
    "# use a count function to enumerate the # of acetylation of K's in each peptide\n",
    "#peaks['k-acet'] = peaks['Peptide'].apply(lambda x: x.count('K(+42.01)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks['stripped_peptide'] = peaks['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks['stripped_length'] = peaks['stripped_peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks['ptm-total'] = peaks['c-carb'] + peaks['m-oxid'] + peaks['n-deam'] + peaks['q-deam'] \n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks['NAAF_num.'] = peaks['Area'] / peaks['stripped_length']\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "peaks.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/processed/TROCAS7_Fusion_Apr2021-all-samples/Apr21-peaks76-DN50-peptide-proc.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaks.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
