{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Manipulation of PeaskDB de novo-assisted database search results of Trocas 7 (April, 2019 high water) lower Amazon river proteomics LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "PeaksDB search results (.csv) of database searches against Henrique's Amazon metagenome (+Hi3)\n",
    "All samples (duplicates of most) included, so `Area` and `Spectral Counts` columns for each injection\n",
    "These were all searched with 15 ppm precursor tolerance and 0.5 ppm fragement ion tolerance\n",
    "Exported at <1.0% FDR\n",
    "\n",
    "Goal:\n",
    "\n",
    "Files with stripped (no PTMs) peptide lists and\n",
    "Columns with #'s of each modification in every sequence\n",
    "Column with stripped peptide lengths (# amino acids)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/amazon/data/TROCAS7_Fusion_Apr2021_PEAKS_76-all-samples\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/amazon/data/TROCAS7_Fusion_Apr2021_PEAKS_76-all-samples/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Apr21-peaks76-DB-peptide.csv           Apr21-peaks76-DB-proteins.fasta\r\n",
      "Apr21-peaks76-DB-protein-peptides.csv  Apr21-peaks76-DB-search-psm.csv\r\n",
      "Apr21-peaks76-DB-proteins.csv          Apr21-peaks76-dno.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Peptide', '-10lgP', 'Mass', 'Length', 'ppm', 'm/z', 'RT',\n",
      "       'Area Trocas7-302-Bay', 'Area Trocas7-306-Chav',\n",
      "       'Area Trocas7-310-SMCP', 'Area Trocas7-318-NMCP',\n",
      "       'Area Trocas7-402-Bay', 'Area Trocas7-406-Chav',\n",
      "       'Area Trocas7-410-SMCP', 'Area Trocas7-417-NMCP',\n",
      "       'Area Trocas7-102-Bay', 'Area Trocas7-106-Chav',\n",
      "       'Area Trocas7-206-Chav', 'Area Trocas7-110-SMCP',\n",
      "       'Area Trocas7-126-NMCP', 'Area Trocas7-202-Bay',\n",
      "       'Area Trocas7-210-SMCP', 'Area Trocas7-410-SMCP-DUP',\n",
      "       'Area Trocas7-226-NMCP', 'Area Trocas7-303-Bay',\n",
      "       'Area Trocas7-310-SMCP-DUP', 'Area Trocas7-102-Bay-DUP',\n",
      "       'Area Trocas7-106-Chav-DUP', 'Area Trocas7-302-Bay-DUP',\n",
      "       'Area Trocas7-306-Chav-DUP', 'Area Trocas7-503-Bay',\n",
      "       'Area Trocas7-519-NMCP', 'Area Trocas7-318-NMCP-DUP',\n",
      "       'Area Trocas7-402-Bay-DUP', 'Area Trocas7-406-Chav-DUP',\n",
      "       'Area Trocas7-417-NMCP-DUP', 'Area Trocas7-307-Chav',\n",
      "       'Area Trocas7-311-SMCP', 'Area Trocas7-319-NMCP',\n",
      "       'Area Trocas7-507-Chav', 'Area Trocas7-511-SMCP', 'Fraction', 'Scan',\n",
      "       'Source File', '#Spec', '#Spec Trocas7-302-Bay',\n",
      "       '#Spec Trocas7-306-Chav', '#Spec Trocas7-310-SMCP',\n",
      "       '#Spec Trocas7-318-NMCP', '#Spec Trocas7-402-Bay',\n",
      "       '#Spec Trocas7-406-Chav', '#Spec Trocas7-410-SMCP',\n",
      "       '#Spec Trocas7-417-NMCP', '#Spec Trocas7-102-Bay',\n",
      "       '#Spec Trocas7-106-Chav', '#Spec Trocas7-206-Chav',\n",
      "       '#Spec Trocas7-110-SMCP', '#Spec Trocas7-126-NMCP',\n",
      "       '#Spec Trocas7-202-Bay', '#Spec Trocas7-210-SMCP',\n",
      "       '#Spec Trocas7-410-SMCP-DUP', '#Spec Trocas7-226-NMCP',\n",
      "       '#Spec Trocas7-303-Bay', '#Spec Trocas7-310-SMCP-DUP',\n",
      "       '#Spec Trocas7-102-Bay-DUP', '#Spec Trocas7-106-Chav-DUP',\n",
      "       '#Spec Trocas7-302-Bay-DUP', '#Spec Trocas7-306-Chav-DUP',\n",
      "       '#Spec Trocas7-503-Bay', '#Spec Trocas7-519-NMCP',\n",
      "       '#Spec Trocas7-318-NMCP-DUP', '#Spec Trocas7-402-Bay-DUP',\n",
      "       '#Spec Trocas7-406-Chav-DUP', '#Spec Trocas7-417-NMCP-DUP',\n",
      "       '#Spec Trocas7-307-Chav', '#Spec Trocas7-311-SMCP',\n",
      "       '#Spec Trocas7-319-NMCP', '#Spec Trocas7-507-Chav',\n",
      "       '#Spec Trocas7-511-SMCP', 'Accession', 'PTM', 'AScore'],\n",
      "      dtype='object')\n",
      "mean peptide length: 11.623059866962306\n",
      "# redundant peaksdb peptides in combined dataframe 902\n",
      "# nonredundant peaksdb peptides in combined dataframe 902\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>-10lgP</th>\n",
       "      <th>Mass</th>\n",
       "      <th>Length</th>\n",
       "      <th>ppm</th>\n",
       "      <th>m/z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area Trocas7-302-Bay</th>\n",
       "      <th>Area Trocas7-306-Chav</th>\n",
       "      <th>Area Trocas7-310-SMCP</th>\n",
       "      <th>...</th>\n",
       "      <th>Area Trocas7-417-NMCP-DUP</th>\n",
       "      <th>Area Trocas7-307-Chav</th>\n",
       "      <th>Area Trocas7-311-SMCP</th>\n",
       "      <th>Area Trocas7-319-NMCP</th>\n",
       "      <th>Area Trocas7-507-Chav</th>\n",
       "      <th>Area Trocas7-511-SMCP</th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LGEHNIDVLEGNEQFINAAK</td>\n",
       "      <td>112.16</td>\n",
       "      <td>2210.0967</td>\n",
       "      <td>20</td>\n",
       "      <td>-2.8</td>\n",
       "      <td>1106.0525</td>\n",
       "      <td>95.57</td>\n",
       "      <td>4370000.0</td>\n",
       "      <td>5920000.0</td>\n",
       "      <td>247000.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18200.0</td>\n",
       "      <td>2030000.0</td>\n",
       "      <td>155000.0</td>\n",
       "      <td>283000.0</td>\n",
       "      <td>1050000.0</td>\n",
       "      <td>112</td>\n",
       "      <td>16192</td>\n",
       "      <td>20210411_Trocas7_668_SMCP311_DDA_120min_1.raw</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SC(+57.02)AAAGTEC(+57.02)LISGWGNTK</td>\n",
       "      <td>104.26</td>\n",
       "      <td>1881.8350</td>\n",
       "      <td>18</td>\n",
       "      <td>2.1</td>\n",
       "      <td>941.9268</td>\n",
       "      <td>93.16</td>\n",
       "      <td>3330000.0</td>\n",
       "      <td>3860000.0</td>\n",
       "      <td>1730000.0</td>\n",
       "      <td>...</td>\n",
       "      <td>570000.0</td>\n",
       "      <td>3370000.0</td>\n",
       "      <td>3140000.0</td>\n",
       "      <td>6150000.0</td>\n",
       "      <td>608000.0</td>\n",
       "      <td>534000.0</td>\n",
       "      <td>111</td>\n",
       "      <td>16381</td>\n",
       "      <td>20210411_Trocas7_667_Chav307_DDA_120min_1.raw</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SSGSSYPSLLQC(+57.02)LK</td>\n",
       "      <td>88.49</td>\n",
       "      <td>1525.7446</td>\n",
       "      <td>14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>763.8811</td>\n",
       "      <td>101.67</td>\n",
       "      <td>371000.0</td>\n",
       "      <td>1150000.0</td>\n",
       "      <td>367000.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>253000.0</td>\n",
       "      <td>178000.0</td>\n",
       "      <td>570000.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>135000.0</td>\n",
       "      <td>111</td>\n",
       "      <td>18330</td>\n",
       "      <td>20210411_Trocas7_667_Chav307_DDA_120min_1.raw</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SGGGGGGGLGSGGSIR</td>\n",
       "      <td>85.72</td>\n",
       "      <td>1231.5905</td>\n",
       "      <td>16</td>\n",
       "      <td>1.9</td>\n",
       "      <td>616.8036</td>\n",
       "      <td>39.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>204000.0</td>\n",
       "      <td>114000.0</td>\n",
       "      <td>270000.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>113</td>\n",
       "      <td>6581</td>\n",
       "      <td>20210411_Trocas7_669_NMCP319_DDA_120min_1.raw</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RHPYFYAPELLFFAKR</td>\n",
       "      <td>83.71</td>\n",
       "      <td>2054.0889</td>\n",
       "      <td>16</td>\n",
       "      <td>2.1</td>\n",
       "      <td>514.5306</td>\n",
       "      <td>131.89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4900.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>81</td>\n",
       "      <td>25155</td>\n",
       "      <td>20210411_Trocas7_666_Bay303_DDA_120min_1.raw</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Peptide  -10lgP       Mass  Length  ppm  \\\n",
       "0                LGEHNIDVLEGNEQFINAAK  112.16  2210.0967      20 -2.8   \n",
       "1  SC(+57.02)AAAGTEC(+57.02)LISGWGNTK  104.26  1881.8350      18  2.1   \n",
       "2              SSGSSYPSLLQC(+57.02)LK   88.49  1525.7446      14  2.0   \n",
       "3                    SGGGGGGGLGSGGSIR   85.72  1231.5905      16  1.9   \n",
       "4                    RHPYFYAPELLFFAKR   83.71  2054.0889      16  2.1   \n",
       "\n",
       "         m/z      RT  Area Trocas7-302-Bay  Area Trocas7-306-Chav  \\\n",
       "0  1106.0525   95.57             4370000.0              5920000.0   \n",
       "1   941.9268   93.16             3330000.0              3860000.0   \n",
       "2   763.8811  101.67              371000.0              1150000.0   \n",
       "3   616.8036   39.30                   NaN                    NaN   \n",
       "4   514.5306  131.89                   NaN                    NaN   \n",
       "\n",
       "   Area Trocas7-310-SMCP  ...  Area Trocas7-417-NMCP-DUP  \\\n",
       "0               247000.0  ...                        NaN   \n",
       "1              1730000.0  ...                   570000.0   \n",
       "2               367000.0  ...                        NaN   \n",
       "3                    NaN  ...                        NaN   \n",
       "4                    NaN  ...                        NaN   \n",
       "\n",
       "   Area Trocas7-307-Chav  Area Trocas7-311-SMCP  Area Trocas7-319-NMCP  \\\n",
       "0                18200.0              2030000.0               155000.0   \n",
       "1              3370000.0              3140000.0              6150000.0   \n",
       "2               253000.0               178000.0               570000.0   \n",
       "3               204000.0               114000.0               270000.0   \n",
       "4                    NaN                    NaN                    NaN   \n",
       "\n",
       "   Area Trocas7-507-Chav  Area Trocas7-511-SMCP  Fraction   Scan  \\\n",
       "0               283000.0              1050000.0       112  16192   \n",
       "1               608000.0               534000.0       111  16381   \n",
       "2                    NaN               135000.0       111  18330   \n",
       "3                    NaN                    NaN       113   6581   \n",
       "4                 4900.0                    NaN        81  25155   \n",
       "\n",
       "                                     Source File  Accession  \n",
       "0  20210411_Trocas7_668_SMCP311_DDA_120min_1.raw        NaN  \n",
       "1  20210411_Trocas7_667_Chav307_DDA_120min_1.raw        NaN  \n",
       "2  20210411_Trocas7_667_Chav307_DDA_120min_1.raw        NaN  \n",
       "3  20210411_Trocas7_669_NMCP319_DDA_120min_1.raw        NaN  \n",
       "4   20210411_Trocas7_666_Bay303_DDA_120min_1.raw        NaN  \n",
       "\n",
       "[5 rows x 45 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_csv function\n",
    "pdb_dup = pd.read_csv(\"/home/millieginty/Documents/git-repos/amazon/data/TROCAS7_Fusion_Apr2021_PEAKS_76-all-samples/Apr21-peaks76-DB-peptide.csv\")\n",
    "\n",
    "# remove redundant rows\n",
    "pdb = pd.DataFrame.drop_duplicates(pdb_dup)\n",
    "\n",
    "print(pdb.columns)\n",
    "\n",
    "##columns = ['Peptide', '-10lgP', 'Mass', 'Length', 'ppm', 'm/z', 'RT',\n",
    "       ##'Area', 'Fraction', 'Scan', 'Source_File',\n",
    "       ##'#Spec', '#Spec', 'Accession', 'PTM',\n",
    "       ##'AScore']\n",
    "\n",
    "##peaksdb322.columns = columns\n",
    "\n",
    "#remmove # spec and accession columns because they mess parsing up\n",
    "\n",
    "##del peaksdb322['#Spec']\n",
    "##del peaksdb322['Accession']\n",
    "del pdb['PTM']\n",
    "del pdb['AScore']\n",
    "\n",
    "# get rid of all the spectral count #s, we're fine with Area\n",
    "pdb = pdb[pdb.columns.drop(list(pdb.filter(regex='Spec')))]\n",
    "\n",
    "mean_length = pdb['Length'].mean()\n",
    "print('mean peptide length:', mean_length)\n",
    "\n",
    "print(\"# redundant peaksdb peptides in combined dataframe\", len(pdb_dup))\n",
    "print(\"# nonredundant peaksdb peptides in combined dataframe\", len(pdb))\n",
    "\n",
    "#look at the dataframe\n",
    "pdb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>-10lgP</th>\n",
       "      <th>Mass</th>\n",
       "      <th>Length</th>\n",
       "      <th>ppm</th>\n",
       "      <th>m/z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area Trocas7-302-Bay</th>\n",
       "      <th>Area Trocas7-306-Chav</th>\n",
       "      <th>Area Trocas7-310-SMCP</th>\n",
       "      <th>...</th>\n",
       "      <th>W</th>\n",
       "      <th>Y</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>stripped_peptide</th>\n",
       "      <th>stripped_length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>stripped_IL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LGEHNIDVLEGNEQFINAAK</td>\n",
       "      <td>112.16</td>\n",
       "      <td>2210.0967</td>\n",
       "      <td>20</td>\n",
       "      <td>-2.8</td>\n",
       "      <td>1106.0525</td>\n",
       "      <td>95.57</td>\n",
       "      <td>4370000.0</td>\n",
       "      <td>5920000.0</td>\n",
       "      <td>247000.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LGEHNIDVLEGNEQFINAAK</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>LGEHNLDVLEGNEQFLNAAK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SC(+57.02)AAAGTEC(+57.02)LISGWGNTK</td>\n",
       "      <td>104.26</td>\n",
       "      <td>1881.8350</td>\n",
       "      <td>18</td>\n",
       "      <td>2.1</td>\n",
       "      <td>941.9268</td>\n",
       "      <td>93.16</td>\n",
       "      <td>3330000.0</td>\n",
       "      <td>3860000.0</td>\n",
       "      <td>1730000.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SCLISGWGNTK</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>SCLLSGWGNTK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SSGSSYPSLLQC(+57.02)LK</td>\n",
       "      <td>88.49</td>\n",
       "      <td>1525.7446</td>\n",
       "      <td>14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>763.8811</td>\n",
       "      <td>101.67</td>\n",
       "      <td>371000.0</td>\n",
       "      <td>1150000.0</td>\n",
       "      <td>367000.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SSGSSYPSLLQCLK</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>SSGSSYPSLLQCLK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SGGGGGGGLGSGGSIR</td>\n",
       "      <td>85.72</td>\n",
       "      <td>1231.5905</td>\n",
       "      <td>16</td>\n",
       "      <td>1.9</td>\n",
       "      <td>616.8036</td>\n",
       "      <td>39.30</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGGGGGGGLGSGGSIR</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>SGGGGGGGLGSGGSLR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RHPYFYAPELLFFAKR</td>\n",
       "      <td>83.71</td>\n",
       "      <td>2054.0889</td>\n",
       "      <td>16</td>\n",
       "      <td>2.1</td>\n",
       "      <td>514.5306</td>\n",
       "      <td>131.89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>RHPYFYAPELLFFAKR</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>RHPYFYAPELLFFAKR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Peptide  -10lgP       Mass  Length  ppm  \\\n",
       "0                LGEHNIDVLEGNEQFINAAK  112.16  2210.0967      20 -2.8   \n",
       "1  SC(+57.02)AAAGTEC(+57.02)LISGWGNTK  104.26  1881.8350      18  2.1   \n",
       "2              SSGSSYPSLLQC(+57.02)LK   88.49  1525.7446      14  2.0   \n",
       "3                    SGGGGGGGLGSGGSIR   85.72  1231.5905      16  1.9   \n",
       "4                    RHPYFYAPELLFFAKR   83.71  2054.0889      16  2.1   \n",
       "\n",
       "         m/z      RT  Area Trocas7-302-Bay  Area Trocas7-306-Chav  \\\n",
       "0  1106.0525   95.57             4370000.0              5920000.0   \n",
       "1   941.9268   93.16             3330000.0              3860000.0   \n",
       "2   763.8811  101.67              371000.0              1150000.0   \n",
       "3   616.8036   39.30                   NaN                    NaN   \n",
       "4   514.5306  131.89                   NaN                    NaN   \n",
       "\n",
       "   Area Trocas7-310-SMCP  ...  W  Y  c-carb  m-oxid  n-deam  q-deam  \\\n",
       "0               247000.0  ...  0  0       0       0       0       0   \n",
       "1              1730000.0  ...  1  0       2       0       0       0   \n",
       "2               367000.0  ...  0  1       1       0       0       0   \n",
       "3                    NaN  ...  0  0       0       0       0       0   \n",
       "4                    NaN  ...  0  2       0       0       0       0   \n",
       "\n",
       "       stripped_peptide  stripped_length  ptm-total           stripped_IL  \n",
       "0  LGEHNIDVLEGNEQFINAAK               20          0  LGEHNLDVLEGNEQFLNAAK  \n",
       "1           SCLISGWGNTK               11          2           SCLLSGWGNTK  \n",
       "2        SSGSSYPSLLQCLK               14          1        SSGSSYPSLLQCLK  \n",
       "3      SGGGGGGGLGSGGSIR               16          0      SGGGGGGGLGSGGSLR  \n",
       "4      RHPYFYAPELLFFAKR               16          0      RHPYFYAPELLFFAKR  \n",
       "\n",
       "[5 rows x 73 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "pdb['A'] = pdb['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "pdb['C'] = pdb['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "pdb['D'] = pdb['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "pdb['E'] = pdb['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "pdb['F'] = pdb['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "pdb['G'] = pdb['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "pdb['H'] = pdb['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in pdb output, there will be no isoleucines (they're lumped in with leucines)\n",
    "pdb['I'] = pdb['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "pdb['K'] = pdb['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "pdb['L'] = pdb['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "pdb['M'] = pdb['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "pdb['N'] = pdb['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "pdb['P'] = pdb['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "pdb['Q'] = pdb['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "pdb['R'] = pdb['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "pdb['S'] = pdb['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "pdb['T'] = pdb['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "pdb['V'] = pdb['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "pdb['W'] = pdb['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "pdb['Y'] = pdb['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "pdb['c-carb'] = pdb['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "pdb['m-oxid'] = pdb['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized K's in each peptide\n",
    "#pdb['k-oxid'] = pdb['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized P's in each peptide\n",
    "#pdb['p-oxid'] = pdb['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized R's in each peptide\n",
    "#pdb['r-oxid'] = pdb['Peptide'].apply(lambda x: x.count('R(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized Y's in each peptide\n",
    "#pdb['y-oxid'] = pdb['Peptide'].apply(lambda x: x.count('Y(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "pdb['n-deam'] = pdb['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "pdb['q-deam'] = pdb['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "#pdb['k-meth'] = pdb['Peptide'].apply(lambda x: x.count('K(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "#pdb['r-meth'] = pdb['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of pyro glu Q's in each peptide\n",
    "#pdb['q-pyro'] = pdb['Peptide'].apply(lambda x: x.count('Q(-17.03)'))\n",
    "\n",
    "# use a count function to enumerate the # of acetylation of K's in each peptide\n",
    "#pdb['k-acet'] = pdb['Peptide'].apply(lambda x: x.count('K(+42.01)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "pdb['stripped_peptide'] = pdb['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "pdb['stripped_length'] = pdb['stripped_peptide'].apply(len)\n",
    "\n",
    "##pdb['NAAF_num.'] = pdb['Area'] / pdb['stripped_length']\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "pdb['ptm-total'] = pdb['c-carb'] + pdb['m-oxid'] + pdb['n-deam'] + pdb['q-deam'] \n",
    "\n",
    "# turn all isoleucines into leucines\n",
    "# this helps later in comparing Unipept peptides to PeaksDB and Comet ones\n",
    "pdb['stripped_IL']= pdb['stripped_peptide'].str.replace('I','L')\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "pdb.to_csv(\"/home/millieginty/Documents/git-repos/amazon/data/processed/TROCAS7_Fusion_Apr2021-all-samples/Apr21-peaks76-DB-peptide-proc.csv\")\n",
    "\n",
    "# check out the results\n",
    "pdb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
